### ISSUE TITLE
How to perform multi-GPU training?

### ISSUE THREAD

**Issue by Mrils** [TRI-ML/packnet-sfm#34]
Hello,
I'm trying to train a model with VelSupModel. And I find out that only single GPU is working. How to perform multi-GPU training? thx.

**Comment by VitorGuizilini-TRI** [comment]
When you say that it doesn't work, do you mean it doesn't converge or training doesn't even start? Our training pipeline by default uses multi-GPU, with Horovod, so there is no "single-GPU" training. If your problem is that it doesn't converge, you might want to look into changing learning rates, velocity supervision loss weight, batch sizes and other training parameters.

**Comment by Mrils** [comment]
I'm afraid not. When I use 'nvidia-smi' to monitor the GPU memory usage, I find that only one GPU (id 0) is working, and when I adjust batch_size to 16, the GPU memory overflowed. I'm new to Horovod,  So I am not sure if something is wrong with my configuration file. Here is my train.yaml
model: Also I am not running in docker environment, but in anaconda virtual env.
    name: 'VelSupModel'
    optimizer:
        name: 'Adam'
        depth:
            lr: 0.0002
        pose:
            lr: 0.0002
    scheduler:
        name: 'StepLR'
        step_size: 30
        gamma: 0.5
    depth_net:
        name: 'DepthResNet'
        version: '50pt'
    pose_net:
        name: 'PoseNet'
        version: ''
    params:
        crop: 'garg'
        min_depth: 0.0
        max_depth: 80.0
datasets:
    augmentation:
        image_shape: (192, 640)
    train:
        batch_size: 16
        dataset: ['KITTI']
        path: ['/datassd/datasets/KITTI_raw']
        split: ['data_splits/eigen_zhou_files.txt']
        depth_type: ['velodyne']
        repeat: [2]
    validation:
        dataset: ['KITTI']
        path: ['/datassd/datasets/KITTI_raw']
        split: ['data_splits/eigen_val_files.txt',
                'data_splits/eigen_test_files.txt']
        depth_type: ['velodyne']
    test:
        dataset: ['KITTI']
        path: ['/datassd/datasets/KITTI_raw']
        split: ['data_splits/eigen_test_files.txt']
        depth_type: ['velodyne']

**Comment by Mrils** [comment]
Sorry, I have figured it out. thank you

**Comment by Mrils** [comment]
Another question: Did you turn off data augmentation during training? As the prob=1.0 in colorjitter_sample function.
[https://github.com/TRI-ML/packnet-sfm/blob/f824ffceba46ae1c621e1bf22a35634d8b39207c/packnet_sfm/datasets/augmentations.py#L197](url)

**Comment by krishna-esrlabs** [comment]
@Mrils Hi, could you please tell me how to run multi-gpu training, I haven't figured out yet.

**Comment by DRAEYE** [comment]
Excuse me, can I ask how do you deal with this problem, on my computer, only one GPU is working

**Comment by abhaysjuneja** [comment]
@Mrils Could you please share your solution on how you fixed the multi-gpu training issue? I'm trying to train in a non-docker environment, and only one GPU is being utilized.

**Comment by jdriscoll319** [comment]
Was anyone able to figure this out?
